src/user-guide/writing-workflows/configuration.rst

```diff
@@ -62,10 +62,137 @@ documentation, configuration files, etc. When the workflow is :ref:`installed
    equivalent directories in the :ref:`workflow_share_directories`.
 
 
+.. _CodeInCylcConfigurations:
+
+Code in Workflow Configurations
+-------------------------------
+
+Cylc workflow configurations are not executable scripts or programs. Rather,
+they configure, at start-up, the Cylc scheduler program to run a particular
+workflow.
+
+The scheduler understands a static syntax that contains only ``[section
+headings]`` and configuration items of the form ``platform = hpc1``. These
+configuration items are not even "variables" that can be referenced elsewhere
+in the file. Some values may look complicated, but they are still just text
+strings with content that is meaningful to the scheduler.
```

Comment:
The tone here is a bit negative, Cylc provides a powerful configuration format, it just isn't dynamic.

Maybe just explain that the Cylc config is static, but can be augmented by template engines such as Jinja2 and explain that these engines are run *before* startup.